README.md

```diff
@@ -4,27 +4,720 @@
 
 Eventually consistent external materialized views.
 
-*Notice: library is still somewhat in flux.*
-
 [![Clojars Project](http://clojars.org/kirasystems/views/latest-version.svg)](http://clojars.org/kirasystems/views)
 
+## Version 2
+
+Version 2.0.0 represents the merging of Gered King's branch with 
+ours. A big thanks to Gered whose changes make the view library 
+much more user friendly and better documented.
+
+## Basic Concepts
+
+The views library allows you to manage a **view system** which is a 
+collection of **views** and a list of **subscribers** to those views. 
+Subscribers will get sent **view refreshes** in realtime when the data 
+represented by the views they are subscribed to changes. Relevant 
+changes are found through the use of **hints** which are added to the 
+view system by anything that is actually changing the data at the 
+instant it is changed.
+
+A **view** is similar in concept to a [materialized view][2], though in
+practice it may not actually keep a copy of the underlying data 
+represented by the view and instead just keep a copy of a query or the 
+location of where the data can be retrieved from when it is needed 
+(e.g. when view refrehses need to be sent out).
+
+[2]: https://en.wikipedia.org/wiki/Materialized_view
+
+A view is represented by the protocol `views.protocols/IView`:
+
+```clj
+(defprotocol IView
+  (data [this namespace parameters])
+  (relevant? [this namespace parameters hints])
+  (id [this]))
+```
+
+`id` simply returns a unique identifier for this view. `data` returns a
+copy of the underlying data represented by this view. `relevant?` 
+determines if a collection of hints are relevant to the view and is 
```

Comment:
The subject here is a "collection of hints", not "hints". To remove ambiguity, you can remove "a collection of" and keep the "are".